backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramCreationService.java

```diff
@@ -79,9 +86,33 @@ public Diagram create(String label, Object targetObject, DiagramDescription diag
         return newDiagram;
     }
 
+    @Override
+    public boolean shouldRefresh(IEditingContext editingContext, IDiagramContext diagramContext, ChangeDescription changeDescription) {
```

Comment:
Why is the DiagramCreationService in charge of the refresh policy of a diagram? I don't see why this code should be in this class.

Reply:
I don't remember well why I put this code here. Maybe because it would have added one parameter to the diagram event processor while the parameter was already used in the diagram creation service and the diagram creation service was already in charge of the refresh, so, why not the should refresh.
Anyway, this code has been moved to the diagram event processor.